File: Data Structures & Algorithms/anagram-groups/submission-2.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        f = {}

        for i in strs:
            freq = [0] * 26;
            for j in i:
                freq[ord(j)-97] += 1
            if tuple(freq) not in f:
                f[tuple(freq)] = []
            f[tuple(freq)].append(i)

        res = []
        for i in f:
            res.append(f[i])
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> freq;

        for (auto i : nums) {
            if (freq[i]) {
                return true;
            }
            else {
                freq[i] = 1;
            }
        }
        return false;
    }
};